t/39-tinycv.t

```diff
@@ -0,0 +1,270 @@
+    ok($olvl eq 0, 'Check 0 for non overlapping small area bottom.');
+
+    undef($olvl);
+    $olvl = tinycv::Image::overlap_lvl($area1, {xpos => 99, ypos => 99, height => 102, width => 102});
+    ok($olvl eq 1, 'Check 1 for fully covering area.');
+
+    undef($olvl);
+    $olvl = tinycv::Image::overlap_lvl($area1, {xpos => 90, ypos => 90, height => 20, width => 120});
+    ok($olvl eq 2, 'Check 2 for area overlapping top edge.');
+
+    undef($olvl);
+    $olvl = tinycv::Image::overlap_lvl($area1, {xpos => 90, ypos => 190, height => 20, width => 120});
+    ok($olvl eq 3, 'Check 3 for area overlapping bottom edge.');
+
+    undef($olvl);
+    $olvl = tinycv::Image::overlap_lvl($area1, {xpos => 90, ypos => 90, height => 120, width => 20});
+    ok($olvl eq 4, 'Check 4 for area overlapping left edge.');
+
+    undef($olvl);
+    $olvl = tinycv::Image::overlap_lvl($area1, {xpos => 190, ypos => 90, height => 120, width => 20});
+    ok($olvl eq 5, 'Check 5 for area overlapping right edge.');
+
+    undef($olvl);
+    $olvl = tinycv::Image::overlap_lvl($area1, {xpos => 90, ypos => 140, height => 20, width => 120});
+    ok($olvl eq 6, 'Check 6 for horizontal split.');
+
+    undef($olvl);
+    $olvl = tinycv::Image::overlap_lvl($area1, {xpos => 140, ypos => 90, height => 120, width => 20});
+    ok($olvl eq 7, 'Check 7 for vertical split.');
+
+    undef($olvl);
+    $olvl = tinycv::Image::overlap_lvl($area1, {xpos => 90, ypos => 90, height => 20, width => 20});
+    ok($olvl eq 8, 'Check 8 for area overlapping top left corner.');
+
+    undef($olvl);
+    $olvl = tinycv::Image::overlap_lvl($area1, {xpos => 190, ypos => 90, height => 20, width => 20});
+    ok($olvl eq 9, 'Check 9 for area overlapping top right corner.');
+
+    undef($olvl);
+    $olvl = tinycv::Image::overlap_lvl($area1, {xpos => 90, ypos => 190, height => 20, width => 20});
+    ok($olvl eq 10, 'Check 10 for area overlapping bottom left corner.');
+
+    undef($olvl);
+    $olvl = tinycv::Image::overlap_lvl($area1, {xpos => 190, ypos => 190, height => 20, width => 20});
+    ok($olvl eq 11, 'Check 11 for area overlapping bottom right corner.');
+
+    undef($olvl);
+    $olvl = tinycv::Image::overlap_lvl($area1, {xpos => 140, ypos => 90, height => 20, width => 20});
+    ok($olvl eq 12, 'Check 12 for partial overlap through top edge.');
+
+    undef($olvl);
+    $olvl = tinycv::Image::overlap_lvl($area1, {xpos => 140, ypos => 190, height => 20, width => 20});
+    ok($olvl eq 13, 'Check 14 for partial overlap through bottom edge.');
+
+    undef($olvl);
+    $olvl = tinycv::Image::overlap_lvl($area1, {xpos => 90, ypos => 140, height => 20, width => 20});
+    ok($olvl eq 14, 'Check 14 for partial overlap through left edge.');
+
+    undef($olvl);
+    $olvl = tinycv::Image::overlap_lvl($area1, {xpos => 190, ypos => 140, height => 20, width => 20});
+    ok($olvl eq 15, 'Check 15 for partial overlap through right edge.');
```

Comment:
That looks rather verbose. Is it really necessary to test each and every variant in all cases? I suggest to focus on the important "corner" cases

Reply:
Maybe just make it a hash with different cases. That would simplifiy it anyway

Reply:
Thank you for the suggestion. I have created a Hash mapping all input parameters to the corresponding output value and test case description.